### ISSUE TITLE
Resetting MainPage causes blank screen to render

### ISSUE THREAD

**Issue by brianlagunas** [xamarin/Xamarin.Forms#2562]
### Description
When you set the Application.MainPage to an instance of a TabbedPage that has a ContentPage on it's Modal nav stack, a blank screen is shown and the ContentPage is no longer on the TabbedPage's modal nav stack after the MainPage property is set.  The hardware back button will then close the app.

### Steps to Reproduce

1. Run the repro app in an emulator
2. Click the Rest button on ViewA
3. Review the code comments to step through and observe the values
4. Observe the blank screen in the emulator

Offending code
```
        private void _button_Clicked(object sender, EventArgs e)
        {
            //build up navigation stack
            var tabbedPage = new TabbedPage();
            tabbedPage.Navigation.PushModalAsync(new ViewB());

            //shoud have 1, this is correct
            var numberOfPages = tabbedPage.Navigation.ModalStack.Count;

            //reset main page - ViewB should be showing, but it's a blank screen
            App.Current.MainPage = tabbedPage;

            //should have 1 page on modal stack as before, but now there is 0
            numberOfPages = tabbedPage.Navigation.ModalStack.Count;
        }
```

### Expected Behavior

ViewB should be correctly added to the TabbedPage's modal nav stack and rendered on the emulator. When the hardware back button is click, the TabbedPage should now be shown and ViewB should be popped off the stack.

### Actual Behavior

A blank screen is show

### Basic Information

- Version with issue: 2.5.0.280555
- Last known good version: NA
- IDE: VS 2017
- Platform Target Frameworks: 
  - iOS:  
  - Android: 
  - UWP:
- Nuget Packages: Xamarin.Forms and it's dependencies

### Screenshots
**App launch**
![image](https://user-images.githubusercontent.com/835562/39386087-29778322-4a30-11e8-9762-aade32ac59d9.png)

**Button Click**
![image](https://user-images.githubusercontent.com/835562/39386109-41836904-4a30-11e8-81ae-50649b866d5d.png)

### Reproduction Link

https://github.com/brianlagunas/MainPageToTabbedPageIssue

**Comment by hartez** [comment]
Looks like setting the MainPage isn't bringing in the new page's modal stack; if you call `tabbedPage.Navigation.PushModalAsync(new ViewB());` _after_ setting the MainPage, it works fine.

**Comment by brianlagunas** [comment]
To give full context, this is the process that the Prism navigation service takes to build up the navigation stack during a deep link.  I will not be able to change the order in which the stack is built, as the entire Prism navigation is built on top of building up the nav stack prior to setting the MainPage.

**Comment by hartez** [comment]
@brianlagunas I didn't mean to suggest that you should change anything - I was just adding some info for whoever gets assigned this issue. AFAICT, what you're doing _should_ work just fine.

**Comment by brianlagunas** [comment]
@hartez Thanks for the update. I was not trying to imply that was your suggestion. I just wanted to give the full context of how the bug was found and the use case surrounding it, just incase "change the way you're doing it" would be suggested later in the conversation.  Thank you for looking into this so quickly.

**Comment by Umar3x** [comment]
This behavior is met when you use some animation to push or pop your pages.

I noticed that when you pushAsync(page, false) or popAsync(false) you won't have a "BlankPage" : 

By the way its not really a blank page, I did not go deep to understand whats going on, but if you popAsync for example to the root page (A masterDetail for example) without animation, you will see a blank page but if you open the menu or do some gestures on the blank page, the page you should normally see will appear. 
Another clue is that if you check your navigation stack, everything's fine, no extra blank page or something like that. That's just an UI Bug with animation on iOS.

**Comment by JoeDurand3** [comment]
Does this need to be retested? I can confirm this is still an issue.

**Comment by dannypease** [comment]
Yes this is still an issue, and quite an important issue when it comes to trying to navigate to a certain page via a push notification for example. Is there a work around at all as I am trying to navigate in the following way:

await _navigationService.NavigateAsync("/TabbedPage?selectedTab=MiddleTab/Page1/Page2");

**Comment by KSemenenko** [comment]
I have the same issue with NavigationPage, any updates?

**Comment by vincentcastagna** [comment]
Can confirm this is still an issue that I meet.

**Comment by MitchBomcanhao** [comment]
i'm running into a similar problem.

App launches and sets a loading page as the MainPage.
as part of displaying that loading page, we run a bit of code to determine where the app should go next.
It may decide to open a masterDetail list page, so it creates it and sets it to MainPage.

scenario a) 
I don't use Device.BeginInvokeOnMainThread when setting these main pages
>> The app gets stuck on the loading page, even if the code has actually set the mainpage to be the masterdetail page.

scenario b) 
I use Device.BeginInvokeOnMainThread when setting mainPage.
>>> the app shows a blank screen instead of the expected masterdetail page.

**Comment by almirvuk** [comment]
Is there any news about this issue? Navigating to some child page after TabbedPage is not possible at all and that is one of the most common scenarios for push notification as @dannypease already mentioned.

**Comment by Digi59404** [comment]
Can confirm - I'm also experiencing this issue.

**Comment by husseinmohkhalil** [comment]
I am facing the same too. still no workaround?

**Comment by KSemenenko** [comment]
the same on XF 4.3, 4.4

**Comment by almirvuk** [comment]
Hi, any news about this one? This is a really big issue for devs who want to build and URI for Prism navigation service which will open some details page from TabbedPage. The real and best usage is when the push notification is received and we want to navigate to a specific page as @dannypease already mentioned.

**Comment by RedasP** [comment]
> the same on XF 4.3, 4.4

What exact version of 4.4 did you use? 
I get this issue on 4.3.0.991250, but not on 4.4.0.991757

**Comment by KSemenenko** [comment]
I also have such a problem.
And I think this is a problem with the big page constructor.
the page is created on the UI thread and if it takes a long time, the application may stick.

**Comment by almirvuk** [comment]
@KSemenenko did you try it with 4.4.0.991757 as @RedasP mentioned?

**Comment by almirvuk** [comment]
Can we get more attention to this one for the next release? @samhouts

**Comment by dansiegel** [comment]
FWIW... this actually affects Hot Reload... as I've been experiencing this when making changes on my initial screen... this really does need some prioritization.

**Comment by samhouts** [comment]
Yep, this is coming back for sprint 168. Thanks for your patience, friends!

**Comment by kingces95** [comment]
I'm seeing this exception when I update to the latest v4. We're investigating but if anyone has an updated reproduction that'd be appreciated. 

> System.TypeLoadException: 'Could not resolve type with token 01000032 from typeref (expected class 'Xamarin.Forms.Xaml.XamlResourceIdAttribute' in assembly 'Xamarin.Forms.Core, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null')'

**Comment by kingces95** [comment]
> Looks like setting the MainPage isn't bringing in the new page's modal stack; if you call tabbedPage.Navigation.PushModalAsync(new ViewB()); after setting the MainPage, it works fine.

Hm, well, it doesn't work on iOS.

**Comment by kingces95** [comment]
After a model has been pushed another page or another model page can be pushed. In this way a stack of partitioned pages is built with each partition representing a model push. Internally we represent this as a list of list with the bottom of the stack being [0][0]. For this feature to work generally, we'd have at allow them to generate this partitioned stack to an arbitrary depth.

**Comment by kingces95** [comment]
Well, this change makes the reproduction work. It's kinda of a stab in the dark. Basically, when assigning the NavigationProxy to Inner that's when the new main page starts reporting its stack to the application. Trouble with the way we had it, the page hasn't yet been crowned the new main page! First, crown the main page, then have it report its stack of pages. Who knows, maybe I'll just pass tests and I can go back to stressing about coronavirus...

![image](https://user-images.githubusercontent.com/4120386/78740063-72ff3200-78f1-11ea-9d79-1a81ccfd68db.png)

**Comment by samhouts** [comment]
I wonder if the null check still needs to be there? Who sets their MainPage to null, though?

**Comment by almirvuk** [comment]
Any news about this one? 🙂

**Comment by AmrAlSayed0** [comment]
I have some update on this issue. I tried to reproduce it with a minimal repo with Xamarin.Forms 4.8.0.1451 but instead of a blank screen I get a ```Xamarin.Forms.InvalidNavigationException: Ancestor must be provided for all pushes except first```.

Here is a minimal repo
[Issues2562.zip](https://github.com/xamarin/Xamarin.Forms/files/5320036/Issues2562.zip)

**Comment by BurkusCat** [comment]
Here is a reproduction project for a related set of issues that the Prism team believes are Xamarin.Forms issues: [TabRepro.zip](https://github.com/xamarin/Xamarin.Forms/files/6103490/TabRepro.zip). The full original ticket is here as to not clutter up this thread: https://github.com/PrismLibrary/Prism/issues/2362

If this is ever fixed, it is probably a good idea to ensure it resolves the issues in this repro project too.
